Mikaela S hand in

mikaelasturk opened the pull request:
https://mikaela-js-project-happy-thoughts.netlify.app/

Update:


I've tried to make this pull request since yesterday evening, something hasnt worked

mikaelasturk committed:
Folder structure/theme/GlobalStyle

mikaelasturk committed:
Installed styled components, added netlify link

mikaelasturk committed:
Added character count and errorhandling to form

mikaelasturk committed:
Like button works, message card showing 'x time ago' Added animation to new messages

mikaelasturk committed:
Cleanup

Nicolinabl commented on src/components/ui/Form.jsx:
```diff
@@ -0,0 +1,61 @@
+export const Form = ({ onSendMessage }) => {
+  const [message, setMessage] = useState('')
+
+  const MIN = 5
+  const MAX = 140
```
Very nice that you added min and max characters allowed!

Nicolinabl commented on src/components/ui/InputField.jsx:
```diff
@@ -0,0 +1,98 @@
+    justify-content: space-between;
+  }
+
+  /* Desktop and up --> */
+  @media (min-width: ${({ theme }) => theme.breakpoints.desktop}) {
+    
+  }
+
+  /* Desktop Large and up --> */
+  @media (min-width: ${({ theme }) => theme.breakpoints.desktopLarge}) {
+    
+  }
```
Nothing inside these media queries, can be deleted?

Nicolinabl commented on src/components/cards/MessageCard.jsx:
```diff
@@ -0,0 +1,62 @@
+  align-items: center;
+  gap: 16px;
+`
+
+const Wrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  gap: 12px;
+`
+
+const timeAgo = (timestamp) => {
+  const seconds = Math.floor((Date.now() - timestamp) / 1000)
+
+  if (seconds < 60) return `${seconds} seconds ago`
+
+  const minutes = Math.floor(seconds / 60)
+  if (minutes < 60) return `${minutes} minutes ago`
+
+  const hours = Math.floor(minutes / 60)
+  if (hours < 24) return `${hours} hours ago`
+
+  const days = Math.floor(hours / 24)
+  if (days < 30) return `${days} days ago`
+
+  const months = Math.floor(days / 30)
+  if (months < 12) return `${months} months ago`
+
+  const years = Math.floor(months / 12)
+  return `${years} years ago`
```
this could have been shortened using moment.js, but nice and clean code anyways!

Nicolinabl commented on src/components/cards/SocialMediaContainer.jsx:
```diff
@@ -0,0 +1,41 @@
+import { LinkedInIcon, GithubIcon, StackOverflowIcon, TwitterIcon, InstagramIcon,  } from "../svg-icons/svg-icons"
+import { SvgIcon } from "../ui/SvgIcon"
+
+const SocialMediaWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: 40px;
+
+  a {
+    color: #D0D0D0;
+    transition: color 0.3s ease;
+
+    &:hover {
+      color: #000000;
+    }
+  }
+`
+
+export const SocialMediaContainer = () => {
+  return (
+    <SocialMediaWrapper>
+      <a href="https://www.linkedin.com/in/yourprofile" target="_blank" rel="noopener noreferrer">
+        <LinkedInIcon />
+      </a>
+      <a href="https://github.com/yourprofile" target="_blank" rel="noopener noreferrer">
+        <SvgIcon width="32" viewBox="0 0 32 32" icon={<GithubIcon />} />
+      </a>
+      <a href="https://stackoverflow.com/users/yourprofile" target="_blank" rel="noopener noreferrer">
+        <StackOverflowIcon />
+      </a>
+      <a href="https://twitter.com/yourprofile" target="_blank" rel="noopener noreferrer">
+        <TwitterIcon />
+      </a>
+      <a href="https://www.instagram.com/yourprofile" target="_blank" rel="noopener noreferrer">
+        <InstagramIcon />
+      </a>  
+    </SocialMediaWrapper>
+  )
+}
```
I like that you added this to the design :)

Nicolinabl commented on src/components/sections/HeroSection.jsx:
```diff
@@ -0,0 +1,16 @@
+import { StyledSection } from '../../styles/StyledSection'
+
+export const HeroSection = ({ variant }) => {
+  return (
```
Also very nice addition to the design!

Nicolinabl commented on src/components/layout/Card.jsx:
```diff
@@ -0,0 +1,6 @@
+export const Card = () => {
```
I see you have a few empty components here. Do they serve a purpose in the finished product?

Nicolinabl commented on src/components/sections/MessageSection.jsx:
```diff
@@ -0,0 +1,17 @@
+      ))}
+    </StyledSection>
+  )
+}
```
Nice and clean!

Nicolinabl commented on src/styles/GlobalStyle.jsx:
```diff
@@ -0,0 +1,218 @@
+import { createGlobalStyle } from "styled-components"
```
Nice and detailed global styles file!

Nicolinabl reviewed (commented):
Nice job!
I like that you followed the design but still implemented some extra details in the design, like the header and contact section, makes it look very fun and welcoming! In general i find your code clean and easy to follow, same with your folder structure. 

I tried to check the requirements, and as far as i could see you follow all apart from using the api. I couldn't find an api fetch anywhere, or other messages from the api displayed on your page. Is it very well hidden somewhere or did you forget to use it?

mikaelasturk committed:
Trying to see if favicon works

mikaelasturk committed:
Added loading on messages

Npahlfer commented on READMETemplate.md:
```diff
@@ -0,0 +1,135 @@
+Describe how the project was implemented, including architectural decisions, state management, and challenges encountered.
+
+### Retrospect
+Summarize reflections, lessons learned, technical insights, and ideas for improvement.
```
No reflections? :)

mikaelasturk committed:
feat: add sidebar hamburger menu component and related navigation updates

- Implemented SidebarHamMenu component for mobile navigation.
- Created index file for navigation components export.
- Added HomePage, LoginPage, MailPage, and SignupPage components.
- Introduced ProtectedRoute for route protection based on user authentication.
- Developed contentStore and thoughtStore for managing application state.
- Added userStore for user authentication state management.
- Included swedishCities.json for city data.
- Updated routes index for new route components.

mikaelasturk committed:
feat: implement user account management features including account update, deletion, and password reset

- Added AccountPage for users to view and edit their account details.
- Implemented ForgotPasswordPage for users to request a password reset link.
- Created ResetPasswordPage for users to set a new password.
- Developed UserProfilePage to display user profiles and manage thoughts.
- Introduced VerifyEmailPage for email verification process.
- Enhanced userStore to persist user authentication state in local storage.
- Updated GlobalStyle to support dark theme.
- Refactored StyledButton and StyledSection for improved styling consistency.
